src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -73,53 +83,131 @@ class VisitService(
   }
 
   fun cancelVisit(offenderNo: String, visitId: Long, visitDto: CancelVisitRequest) {
+    val today = LocalDate.now()
+    val visit = visitRepository.findById(visitId).orElseThrow(PrisonerNotFoundException(offenderNo))
+    val visitOutcome = visitOutcomeRepository.findById(VisitOutcomeReason.pk(visitDto.outcome)).orElseThrow()
+
+    val cancVisitStatus = visitStatusRepository.findById(VisitStatus.CANCELLED).orElseThrow()
+    val absEventOutcome = eventOutcomeRepository.findById(EventOutcome.ABS).orElseThrow()
+    val cancEventStatus = eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow()
```

Comment:
just curious - what is ABS?

Reply:
"Absence" - this is just what Paul Ms PVB code did